crates/ruff_python_semantic/src/scope.rs

```diff
@@ -66,6 +70,7 @@ impl<'a> Scope<'a> {
             shadowed_bindings: FxHashMap::default(),
             globals_id: None,
             flags: ScopeFlags::empty(),
+            unresolved_references: UnresolvedReferences::default(),
```

Comment:
The reason for adding unresolved references to scope is that having a global unresolved reference won't work if there are two attributes with the same name in a program.

Imagine there are two classes and one of them has unresolved attribute `x`. We add this attribute name to the unresolved references and the lint rule runs in both classes. In the first class the rule emits error which is fine. But when the lint rule runs on the second class, it iterates over the unresolved references again and emits the error, because the x was in the unresolved references and it's defined.

```
class A:
    def __init__(self):
        print(self.x)
	self.x = 1

class B:
    def __init__(self):
        self.x = 1
	print(self.x)
```

To avoid this I think having the ast information about the unresolved reference will resolve the confusion. But that requires changing the API. So I proceeded with a flag and added another unresolved references to each scope which currently only contains attributes.